src/Catalog/Persistence/AzureStorage.cs

```diff
@@ -226,7 +226,15 @@ protected override async Task OnSaveAsync(Uri resourceUri, StorageContent conten
 
                     destinationStream.Seek(0, SeekOrigin.Begin);
 
-                    await blob.UploadFromStreamAsync(destinationStream, cancellationToken);
+                    var accessCondition = (content as StringStorageContentWithAccessCondition)?.AccessCondition 
+                        ?? AccessCondition.GenerateEmptyCondition();
+
+                    await blob.UploadFromStreamAsync(
+                        destinationStream, 
+                        accessCondition, 
+                        new BlobRequestOptions(), 
```

Comment:
I think these can be null instead of empty

Reply:
done